otter/integration/lib/nova.py

```diff
@@ -83,3 +90,49 @@ def delete_servers(server_ids, rcs, pool=None, _treq=treq):
     """
     return gatherResults([NovaServer(id=_id, pool=pool, treq=_treq).delete(rcs)
                           for _id in server_ids])
+
+
+def list_servers(rcs, pool, _treq=treq):
+    """
+    Get a list of all servers, with an optional name regex provided.  This
+    does not handle pagination, and instead just increases the limit to an
+    absurdly high number.
+    """
+    params = {'limit': 10000}
+    return _treq.get(
+        "{}/servers/detail".format(rcs.endpoints['nova']),
+        params=params,
+        headers=headers(str(rcs.token)),
+        pool=pool
+    ).addCallback(check_success, [200]).addCallback(_treq.json_content)
+
+
+def wait_for_servers(rcs, pool, group, matcher, timeout=600, period=10,
+                     clock=None, _treq=treq):
+    """
+    Wait until Nova reaches a particular state with regards to the servers for
+    the given group.
+    """
+    @inlineCallbacks
+    def do_work():
+        servers = yield list_servers(rcs, pool, _treq=_treq)
+        servers_in_group = [
+            server for server in servers['servers']
+            if group.group_id in (
+                server['metadata'].get("rax:autoscale:group:id", None),
+                server['metadata'].get("rax:auto_scaling_group_id", None))
```

Comment:
Is there a reason you need to check both? I'd rather just check the "new" one, and stop assigning the old one for convergence tenants.